packages/ember-runtime/lib/mixins/container_proxy.js

```diff
@@ -55,7 +55,14 @@ export default Mixin.create({
    @param {Object} options
    @return {any}
    */
-  lookup: containerAlias('lookup'),
+  lookup() {
+    let container = get(this, '__container__');
+    let result = container.lookup(...arguments);
+    if (result && result.owner === undefined) {
+      result.owner = this;
```

Comment:
I assume from the syntax highlighting that you mean `__container__`, right?

My concern with using anything related to "container" publicly is that it breaks the clean encapsulation. Devs will continue to think of `Container` as semi-private and try to dig into `container.registry` for instance.

Maybe `__owner__`? It's probably safe from collisions, clearly private, and clearly not an instance of a `Container`. I guess I've been unclear how private / public we want this to appear (given that `container` has been available without dunderscores).